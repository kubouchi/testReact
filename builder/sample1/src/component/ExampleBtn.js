import React from 'react'

export default class ExampleBtn extends React.Component {
    render () {
      return (
        <button
          type={this.props.type}
        >
          {this.props.children}
        </button>
      )
    }
  }