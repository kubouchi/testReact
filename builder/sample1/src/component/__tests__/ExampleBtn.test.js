import React from 'react'
import ExampleBtn from '../ExampleBtn'
import renderer from 'react-test-renderer'

test('render test', () => {
  const component = renderer.create(
    <ExampleBtn type="text">Test</ExampleBtn>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('render without type props', () => {
  const component = renderer.create(
    <ExampleBtn>Test</ExampleBtn>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})