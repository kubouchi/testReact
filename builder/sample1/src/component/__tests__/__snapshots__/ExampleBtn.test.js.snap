// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render test 1`] = `
<button
  type="text"
>
  tttttttt
  Test
</button>
`;

exports[`render without type props 1`] = `
<button>
  tttttttt
  Test
</button>
`;
